Remove Backends submodule, Change interface to create RDataFrame

vepadulano opened the pull request:
The new interface to create a distributed RDF follows:
```
from PyRDF import Spark

df = Spark.RDataFrame(...)

type(df) # PyRDF.DataFrame.RDataFrame
```

Files have been moved / renamed accordingly.

etejedor commented on PyRDF/Spark/Backend.py:
```diff
@@ -2,8 +2,8 @@
 from PyRDF import Node
-from PyRDF.Backends import Base
-from PyRDF.Backends import Utils
+from PyRDF import Base
```
I don't think you need to remove `Backends` from `PyRDF`. The internal structure that you have in `PyRDF` does not affect what will appear as user interface in `ROOT.RDF.Distributed`, since there we will do a dynamic injection of what we want where we want. Having `Backends` in `PyRDF` keeps things more tidy.

vepadulano replied:
I see. I just thought I would mirror all changes that would go into ROOT. I'll change this quickly

etejedor replied:
My point is, the internal structure of `PyRDF` can stay as we want, whatever makes more sense. What the user will see is not that, but what we want them to see in `ROOT.RDF.Distributed`.

etejedor commented on PyRDF/Spark/__init__.py:
```diff
@@ -1,16 +1,10 @@
-
-
-def make_spark_dataframe(*args, **kwargs):
+def RDataFrame(*args, **kwargs):
```
It's ok to do it like this for now, but when we integrate this into ROOT and we do the injection, we'll recheck how we want to do such injection.

etejedor commented on PyRDF/__init__.py:
```diff
@@ -4,7 +4,12 @@
+# TODO
+# No dynamic injection of all submodules for now because that would include
+# also Base, Operation, Node etc. which we think should be private. Once we
+# will change their names with leading underscores those would not clutter the
```
We might not even need to change the names of these modules with the underscore to make the private, because in the end the user will never import `PyRDF` (or `DistRDF`, as we could rename it). The user will import stuff in `ROOT.RDF.Distributed`, and that stuff will be injected dynamically from stuff coming from `DistRDF`, which will be the only user-visible interface.

vepadulano committed:
Remove Backends submodule, Change interface to create RDataFrame

The new interface to create a distributed RDF follows:
```
from PyRDF import Spark

df = Spark.RDataFrame(...)

type(df) # PyRDF.DataFrame.RDataFrame
```
Files have been moved / renamed accordingly.
